Services/Mail/classes/AutoResponder/class.ilAutoResponderRepository.php

```diff
@@ -0,0 +1,27 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ ********************************************************************
+ */
+
+interface ilAutoResponderRepository
+{
+    public function findBySenderId(int $sender_id) : array;
+    public function findByReceiverId(int $receiver_id) : array;
```

Comment:
Please use `PHPDocs` to describe the shape of the array. Or event better. Don't use an array at all, but instead create a `AutoResponderCollection` (or use a similar/better name).